src/main.cpp

```diff
@@ -1240,37 +1240,35 @@ bool AcceptToMemoryPoolWorker(
         std::vector <uint256> &vHashTxnToUncache,
         bool isCheckWalletTransaction,
         bool markZcoinSpendTransactionSerial) {
+    bool fTestNet = (Params().NetworkIDString() == CBaseChainParams::TESTNET);
+    LogPrintf("AcceptToMemoryPoolWorker(),fCheckInputs=%s, tx.IsZerocoinSpend()=%s, fTestNet=%s\n",
+              fCheckInputs, tx.IsZerocoinSpend() || tx.IsZerocoinSpendV3(), fTestNet);
+    uint256 hash = tx.GetHash();
     AssertLockHeld(cs_main);
+    if (pfMissingInputs)
+        *pfMissingInputs = false;
+
+    auto& consensus = Params().GetConsensus();
+
     if (tx.IsZerocoinMint()) {
         // Shows if old zerocoin mints are allowed yet in the mempool.
-        bool V2ZerocoinMintsAllowedInMempool = (chainActive.Height() <= Params().GetConsensus().nMintV3SigmaStartBlock + Params().GetConsensus().nMintV2MempoolGracefulPeriod);
+        bool allow = (chainActive.Height() <= consensus.nSigmaStartBlock + consensus.nZerocoinV2MintMempoolGracefulPeriod);
 
-        if (!V2ZerocoinMintsAllowedInMempool)
+        if (!allow) {
             return state.DoS(100, error("Old zerocoin mints no more allowed in mempool"),
                              REJECT_INVALID, "bad-txns-zerocoin");
+        }
     }
+
     if (tx.IsZerocoinSpend()) {
         // Shows if old zerocoin spends are allowed yet in the mempool.
-        bool V2ZerocoinSpendsAllowedInMempool = (chainActive.Height() <= Params().GetConsensus().nZerocoinV2SpendStopBlock);
+        bool allow = (chainActive.Height() <= consensus.nSigmaStartBlock + consensus.nZerocoinV2SpendMempoolGracefulPeriod);
 
-        if (!V2ZerocoinSpendsAllowedInMempool)
+        if (!allow) {
             return state.DoS(100, error("Old zerocoin spends no more allowed in mempool"),
                              REJECT_INVALID, "bad-txns-zerocoin");
+        }
     }
-    if (tx.IsZerocoinSpendV3() || tx.IsZerocoinMintV3()) {
```

Comment:
By removing these checks you allowed both sigma Spend and mint in previous blocks. Maybe we just agree to set the nSigmaStartBlock to a block in the past, and increate zerocoin mint gracefull period for testnet? And not touch the code for mainnet?

Reply:
I think logic here do not allow to accept the block that we have a problem with. the logic that effect to the block is in `ConnectBlock` and I agree to accept the block even the block number is lower than `nSigmaStartBlock` because it may have some script like this in mainnet which effect us to stuck at some block.